registry/client/repository.go

```diff
@@ -604,6 +621,45 @@ func (bs *blobs) Resume(ctx context.Context, id string) (distribution.BlobWriter
 	panic("not implemented")
 }
 
+func (bs *blobs) Mount(ctx context.Context, sourceRepo string, dgst digest.Digest) (distribution.Descriptor, error) {
```

Comment:
Yeah, I got confused about this. I remembered refactoring everything to use the reference package in #1270, but it turns out that didn't get merged and I closed it until other PRs got merged. I'll reopen that PR and bring it up to date, but this PR can go in as-is first.